controllers/helmchartproxy/helmchartproxy_controller.go

```diff
@@ -197,6 +209,102 @@ func (r *HelmChartProxyReconciler) reconcileNormal(ctx context.Context, helmChar
 		}
 	}
 
+	// RolloutStepSize is defined; check if count of HelmReleaseProxies matches
+	// the count of matching clusters.
+	if helmChartProxy.Spec.RolloutStepSize != nil {
+		if len(clusters) != len(helmReleaseProxies) {
+			// Set HelmReleaseProxiesRolloutCompletedCondition to false as
+			// HelmReleaseProxies are being rolled out.
+			conditions.MarkFalse(
+				helmChartProxy,
+				addonsv1alpha1.HelmReleaseProxiesRolloutCompletedCondition,
+				addonsv1alpha1.HelmReleaseProxiesRolloutNotCompleteReason,
+				clusterv1.ConditionSeverityInfo,
+				"%d Helm release proxies not yet rolled out",
+				len(clusters)-len(helmReleaseProxies),
+			)
+
+			// Identifies clusters by their NamespacedName and gathers their
+			// helmReleaseProxyRolloutMeta.
+			clusterRolloutMeta := map[string]*helmReleaseProxyRolloutMeta{}
+			for _, cls := range clusters {
+				clusterRolloutMeta[types.NamespacedName{Namespace: cls.Namespace, Name: cls.Name}.String()] = &helmReleaseProxyRolloutMeta{cluster: cls}
+			}
+			for _, hrp := range helmReleaseProxies {
+				hrpClsRef := hrp.Spec.ClusterRef
+				rltMeta := clusterRolloutMeta[types.NamespacedName{Namespace: hrpClsRef.Namespace, Name: hrpClsRef.Name}.String()]
+				rltMeta.hrpExists = true
+			}
+
+			hrpReadyCond := conditions.Get(helmChartProxy, addonsv1alpha1.HelmReleaseProxiesReadyCondition)
+			// If HelmReleaseProxiesReadyCondition is false, reconcile existing
+			// HelmReleaseProxies and exit.
+			if hrpReadyCond != nil && (hrpReadyCond.Status == corev1.ConditionFalse) {
```

Comment:
Can you utilize conditions.IsTrue() or conditions.IsFalse() here and later in the code.